### PR TITLE
Cleanup: Span and Store

### PR DESCRIPTION
**Pull request by masaori335**
No logic change.

1. Remove unused functions and members from `struct Span`
2. Rename `disk` to `span`
3. Include only used headers

### CODE REVIEW

File: src/iocore/cache/Store.cc [hunk 1]
```diff
@@ -486,7 +484,6 @@ Span::init(const char *path, int64_t size)
   }
 
   // A directory span means we will end up with a file, otherwise, we get what we asked for.
-  this->set_mmapable(ink_file_is_mmappable(S_ISDIR(sbuf.st_mode) ? static_cast<mode_t>(S_IFREG) : sbuf.st_mode));
```
Comment (masaori335) [inline src/iocore/cache/Store.cc:489]: This is setting `Span::is_mmapable_internal`, but it's used nowhere.

**Comment by oknet** [comment]
If we don't need to recall the old `MultiCache`, then these cleanups are OK.

File: include/iocore/cache/Store.h [hunk 2]
```diff
@@ -189,25 +141,23 @@ struct Store {
...
       }
-      n_disks = i;
+      n_spans = i;
     }
   }
 
   void delete_all();
 
-  Store();
+  Store(){};
```
Comment (bneradt) [inline include/iocore/cache/Store.h:155]: Suggest: `Store() = default;`

But not a big deal.

**Review by bneradt — APPROVED** [review]
Looks good.

File: include/iocore/cache/Store.h (hunk 2 above)
Comment (masaori335) [reply include/iocore/cache/Store.h:155]: I'll cleanup later.

**Pull request merged**